### ISSUE TITLE
View queued builds

### ISSUE THREAD

**Issue by mdub** [buildkite/feedback#225]
I want to be able to:

* view queued builds for a specific queue
* summarise queued builds across all queues

**Comment by keithpitt** [comment]
@mdub totally! This has been on our TODO list for ages, we just gotta get around to it. What sort of problems are you having when you say "damn, I wish BK had a way for you to view queued builds for a specific queue"? I'd be keen to hear what you're trying to debug at the time!

**Comment by mdub** [comment]
Well, I was planning to shut down an agent cluster, and wanted to see what (and how many) projects were affected.

**Comment by jufemaiz** [comment]
Was going to say something similar regarding global build list (with filtering by state etc).

One problem though @mdub is that you may not know what agent dependencies exist until run time.

**Comment by mdub** [comment]
@jufemaiz: well, I'm not really concerned about finding the pipelines that are configured to use a given queue ... only the builds that are _currently waiting_ to run.

**Comment by jufemaiz** [comment]
Agreed @mdub. But you may well have a build that is currently waiting to run, which starts on a given agent, but makes use of other agents, however the awareness of it isn't available until after the build commences and the pipeline.yml file is accessed by the agent to see the steps taken and the agents used there.

Just an edge case to the above.

**Comment by mdub** [comment]
Fair point. I guess it's really _jobs_ I'm concerned with, rather than builds.

**Comment by mdub** [comment]
A workmate has pointed out that this feature already exists!

* https://buildkite.com/organizations/SLUB/jobs

It's not particularly discoverable, though ... based on the fact I couldn't find that page myself, when browsing in the UI.

**Comment by keithpitt** [comment]
@mdub ah yeah, that's a secret page :) We use it mostly for debugging and let customers know about it when they need to do top level querying like that. It's not 100% ready for prime time, and it's a little bit slow on larger accounts - but it's definitely there if you need it!

**Comment by pauldraper** [comment]
> damn, I wish BK had a way for you to view queued builds for a specific queue

As a very basic thing, I want to know how backed up things are or not, and where my job is. It is gonna finish 5 minutes, or 5 hours? Are we there yet?????

(Per queue is relevant insofar as answering that question depends on the queue if multiple queues are being used.)

There's no great way to see the equivalent of the left side of Jenkins: https://issues.jenkins-ci.org/secure/attachment/32833/build-queue.png Thanks for the tip on /jobs

**Comment by or-shachar** [comment]
+1 on this

**Comment by 0xcaff** [comment]
The endpoint powering the https://buildkite.com/organizations/SLUB/jobs no longer works.

This query times out. Removing agentQueryRules lets it complete but doesn't allow for filtering.

```
query JobsRefetchQuery {
  organization(slug: "coinbase") {
    ...Jobs_organization_2D55O6
    id
  }
}

fragment Jobs_organization_2D55O6 on Organization {
  name
  slug
  jobs(first: 10, type: COMMAND, agentQueryRules: []) {
    edges {
      node {
        __typename
        ... on JobTypeCommand {
          id
        }
        ...Row_job
        ... on Node {
          id
        }
      }
    }
    pageInfo {
      hasNextPage
    }
  }
}

fragment Row_job on Job {
  ... on JobTypeCommand {
    id
    state
    priority {
      number
    }
    agentQueryRules
    passed
    concurrency {
      group
      limit
    }
    createdAt
  }
  ...JobLink_job
}

fragment JobLink_job on JobTypeCommand {
  ...JobTitle_job
  url
}

fragment JobTitle_job on JobTypeCommand {
  label
  command
  build {
    number
    pipeline {
      name
      id
    }
    id
  }
}

```

**Comment by 0xcaff** [comment]
Nvm it sorta works if you use this url.

https://buildkite.com/organizations/coinbase/jobs?q=queue=macos
